sqlglot/dialects/clickhouse.py

```diff
@@ -1106,3 +1106,9 @@ def replacepartition_sql(self, expression: exp.ReplacePartition) -> str:
 
         def projectiondef_sql(self, expression: exp.ProjectionDef) -> str:
             return f"PROJECTION {self.sql(expression.this)} {self.wrap(expression.expression)}"
+
+        def except_op(self, expression: exp.Except) -> str:
+            return f"EXCEPT{' DISTINCT' if expression.args.get('distinct') else ' ALL'}"
```

Comment:
I went with rendering the explicit all for clarity in the generated SQL even though it isn't strictly necessary for clickhouse, rather than minimizing the SQL generated, since effectively every other dialect of SQL considers `EXCEPT` without `ALL` to mean `DISTINCT`. It's very confusing to read the output SQL without `ALL`, and have the execution mean `ALL`.

Reply:
Yeah, good call. Take a look at https://github.com/tobymao/sqlglot/pull/4007#discussion_r1736255398 as well.